cpp/shell.nix

```diff
@@ -0,0 +1,11 @@
+{ pkgs ? import <nixpkgs> {} } : pkgs.mkShell {
+	buildInputs = with pkgs; [
+		cmake
+		gnumake
+		SDL2
+		
+		hostname
+		
+		clang # Massive, and only used for format.sh. But I think it may share some dependencies with Nim, Zig (11/generic, I think), Rust. Painful because the other requirements are so small.
```

Comment:
It probably does not matter since — as far as I know — `clang-format` isn't influenced by the usual compiler flags but I think it's suggested to use the clang tools from [`clang-tools`](https://search.nixos.org/packages?channel=22.11&show=clang-tools&from=0&size=50&sort=relevance&type=packages&query=clang-tools) instead of `clang`.

Doesn't help with the closure size, unfortunately.